src/helpers/ConfigHelper.ts

```diff
@@ -33,13 +40,186 @@ export interface IntegrationCapabilities {
+        categories: slidedownConfig?.categories?.tags
+      }]
+    } as SlidedownOptions;
+  }
+
+  /**
+   * For use with Custom Code & Wordpress Implementations
+   * The OneSignal Wordpress Plugin still uses these legacy keys to set the slidedown text
+   * @param  {any} slidedownConfig
+   * @returns boolean
+   */
+  private static isPromptOptionsVersion0(slidedownConfig: any) : boolean {
+    if (!!slidedownConfig) {
+      const version0Keys = [
+        'acceptButtonText',
+        'cancelButtonText',
+        'actionMessage'
+      ];
+
+      for (let i = 0; i < version0Keys.length; i++) {
+        if (slidedownConfig.hasOwnProperty(version0Keys[i])) return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * For use with Custom Code Implementations
+   * Checks whether `slidedownConfig` implements `SlidedownOptionsVersion1` interface
+   * ------------------------------
+   * v1 schema:
+   * ----------
+   * "slidedown": {
+   *    "enabled": true,
+   *    "autoPrompt": true,
+   *    "...",
+   *    "categories": {...}
+   * }
+   *
+   * v2 schema:
+   * ----------
+   * "slidedown": {
+   *    "prompts": [{...}, {...}, {...}]
+   * }
+   *
+   * Since config can also be set via custom-code and we have no strict checks,
+   * this function helps to check whether the config implements any v1 style config options
+   * by looking for any of the v1 payload first-level keys. See `SlidedownOptionsVersion1`
+   * for the full list of keys.
+   * @param slidedownConfig
+   */
+  private static isSlidedownConfigVersion1(
+    slidedownConfig: any) : slidedownConfig is SlidedownOptionsVersion1 {
+      const version1Keys = [
+        'enabled',
+        'autoPrompt',
+        'pageViews',
+        'timeDelay',
+        'acceptButton',
+        'acceptButtonText',
+        'cancelButton',
+        'cancelButtonText',
+        'actionMessage',
+        'customizeTextEnabled',
+        'categories'
+      ];
+
+      for (let i = 0; i < version1Keys.length; i++) {
+        if (slidedownConfig.hasOwnProperty(version1Keys[i])) return true;
+      }
+
+      return false;
+    }
+
   public static async getAppConfig(userConfig: AppUserConfig,
     downloadServerAppConfig: (appId: string) => Promise<ServerAppConfig>): Promise<AppConfig> {
     try {
       if (!userConfig || !userConfig.appId || !OneSignalUtils.isValidUuid(userConfig.appId))
         throw new SdkInitError(SdkInitErrorKind.InvalidAppId);
 
       const serverConfig = await downloadServerAppConfig(userConfig.appId);
+      const isCustomCode = this.getConfigIntegrationKind(serverConfig) === ConfigIntegrationKind.Custom;
+      const isWordpress  = this.getConfigIntegrationKind(serverConfig) === ConfigIntegrationKind.WordPress;
+
+      if (this.isPromptOptionsVersion0(userConfig.promptOptions)) {
+        userConfig.promptOptions = this.convertConfigToVersionOne(userConfig.promptOptions);
+      }
+
+      if (isCustomCode && !!userConfig.promptOptions?.slidedown) {
```

Comment:
We should remove the `isCustomCode`, we should convert based on format alone. We do know that the format from server will be but the check isn't needed so it will simplify the code.